Add link to profiling recipe from rpc main docs

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45235 Add link to profiling recipe from rpc main docs**

This is so that users know that the profiler works as expected with
RPC and they can learn how to use it to profile RPC-based workloads.

Screenshot:
![profiler_ss](https://user-images.githubusercontent.com/8039770/94075767-a145d080-fdb0-11ea-872a-f88a997c0b2a.png)

Differential Revision: [D23777888](https://our.internmc.facebook.com/intern/diff/D23777888/)

rohan-varma committed:
Add link to profiling recipe from rpc main docs

This is so that users know that the profiler works as expected with
RPC and they can learn how to use it to profile RPC-based workloads.

Differential Revision: [D23777888](https://our.internmc.facebook.com/intern/diff/D23777888/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Add link to profiling recipe from rpc main docs"


This is so that users know that the profiler works as expected with
RPC and they can learn how to use it to profile RPC-based workloads.

Differential Revision: [D23777888](https://our.internmc.facebook.com/intern/diff/D23777888/)

[ghstack-poisoned]

rohan-varma commented:
![profiler_ss](https://user-images.githubusercontent.com/8039770/94075767-a145d080-fdb0-11ea-872a-f88a997c0b2a.png)

mrshenli commented on docs/source/rpc.rst:
```diff
@@ -293,8 +293,11 @@ The RRef design note covers the design of the :ref:`rref` (Remote REFerence) pro
-The RPC tutorial introduces users to the RPC framework and provides two example applications using :ref:`torch.distributed.rpc<distributed-rpc-framework>` APIs.
+The RPC tutorials introduce users to the RPC framework, provide several example applications
+using :ref:`torch.distributed.rpc<distributed-rpc-framework>` APIs, and demonstrate how
+to use `the profiler <https://pytorch.org/docs/stable/autograd.html>`__ to profile RPC-based workloads.
```
Will the link below be more accurate?

```
https://pytorch.org/docs/stable/autograd.html#profiler
```

rohan-varma replied:
Yes, thanks, will update!

mrshenli reviewed (approved):
Thanks for adding this!

rohan-varma committed:
Update on "Add link to profiling recipe from rpc main docs"


This is so that users know that the profiler works as expected with
RPC and they can learn how to use it to profile RPC-based workloads.

Screenshot:
![profiler_ss](https://user-images.githubusercontent.com/8039770/94075767-a145d080-fdb0-11ea-872a-f88a997c0b2a.png)

Differential Revision: [D23777888](https://our.internmc.facebook.com/intern/diff/D23777888/)

[ghstack-poisoned]
